core/src/test/java/greencity/controller/PlaceControllerTest.java

```diff
@@ -606,4 +580,126 @@ void getAllPlaces() throws Exception {
+        verify(restClient, times(1))
+            .sendEmailNotificationChangesPlaceStatus(any(UpdatePlaceStatusWithUserEmailDto.class));
+    }
+
+    @Test
+    void updateStatusDoesNotSendNotificationButSavesDataWhenStatusIsPROPOSED() throws Exception {
+        UpdatePlaceStatusWithUserEmailDto dto = new UpdatePlaceStatusWithUserEmailDto();
+        dto.setPlaceName("testPlace");
+        dto.setNewStatus(PlaceStatus.PROPOSED);
+        dto.setUserName("testUser");
+        dto.setEmail("user@example.com");
+        Place place = new Place();
+        place.setId(1L);
+        place.setName("testPlace");
+        place.setStatus(PlaceStatus.PROPOSED);
+        when(placeService.updatePlaceStatus(any(UpdatePlaceStatusWithUserEmailDto.class))).thenReturn(dto);
+        String json = """
+                {
+                    "placeName": "testPlace",
+                    "newStatus": "PROPOSED",
+                    "userName": "testUser",
+                    "email": "user@example.com"
+                }
+            """;
+
+        mockMvc.perform(patch(placeLink + "/status")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(json))
+            .andExpect(status().isOk())
+            .andExpect(content().string("Status updated successfully for place: testPlace"));
+        verify(placeService, times(1)).updatePlaceStatus(any(UpdatePlaceStatusWithUserEmailDto.class));
+        verify(restClient, times(0))
+            .sendEmailNotificationChangesPlaceStatus(any(UpdatePlaceStatusWithUserEmailDto.class));
+    }
+
+    @Test
+    void updateStatusDoesNotSendNotificationButSavesDataWhenStatusIsDELETED() throws Exception {
+        UpdatePlaceStatusWithUserEmailDto dto = new UpdatePlaceStatusWithUserEmailDto();
+        dto.setPlaceName("testPlace");
+        dto.setNewStatus(PlaceStatus.DELETED);
+        dto.setUserName("testUser");
+        dto.setEmail("user@example.com");
+        Place place = new Place();
+        place.setId(1L);
+        place.setName("testPlace");
+        place.setStatus(PlaceStatus.DELETED);
+        when(placeService.updatePlaceStatus(any(UpdatePlaceStatusWithUserEmailDto.class))).thenReturn(dto);
+        String json = """
+                {
+                    "placeName": "testPlace",
+                    "newStatus": "DELETED",
+                    "userName": "testUser",
+                    "email": "user@example.com"
+                }
+            """;
+
+        mockMvc.perform(patch(placeLink + "/status")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(json))
+            .andExpect(status().isOk())
+            .andExpect(content().string("Status updated successfully for place: testPlace"));
+        verify(placeService, times(1)).updatePlaceStatus(any(UpdatePlaceStatusWithUserEmailDto.class));
+        verify(restClient, times(0))
+            .sendEmailNotificationChangesPlaceStatus(any(UpdatePlaceStatusWithUserEmailDto.class));
+    }
```

Comment:
4 tests looks similar except status and that APPROVED, DECLINED send email notification about changes places status and PROPOSED, DELTED don't. mb, would be better to use parameterized tests, imho